Mathlib/RingTheory/AdicCompletion/Algebra.lean

```diff
@@ -50,20 +50,33 @@ def transitionMapₐ {m n : ℕ} (hmn : m ≤ n) :
     R ⧸ (I ^ n • ⊤ : Ideal R) →ₐ[R] R ⧸ (I ^ m • ⊤ : Ideal R) :=
   AlgHom.ofLinearMap (transitionMap I R hmn) rfl (transitionMap_map_mul I hmn)
 
-/-- `AdicCompletion I R` is an `R`-subalgebra of `∀ n, R ⧸ (I ^ n • ⊤ : Ideal R)`. -/
-def subalgebra : Subalgebra R (∀ n, R ⧸ (I ^ n • ⊤ : Ideal R)) :=
-  Submodule.toSubalgebra (submodule I R) (fun _ ↦ by simp)
-    (fun x y hx hy m n hmn ↦ by simp [hx hmn, hy hmn])
-
-/-- `AdicCompletion I R` is a subring of `∀ n, R ⧸ (I ^ n • ⊤ : Ideal R)`. -/
-def subring : Subring (∀ n, R ⧸ (I ^ n • ⊤ : Ideal R)) :=
-  Subalgebra.toSubring (subalgebra I)
-
-instance : CommRing (AdicCompletion I R) :=
-  inferInstanceAs <| CommRing (subring I)
-
-instance : Algebra R (AdicCompletion I R) :=
-  inferInstanceAs <| Algebra R (subalgebra I)
+instance : Mul (AdicCompletion I R) where
```

Comment:
Can you make the data instances `irreducible`?

Reply:
Done.